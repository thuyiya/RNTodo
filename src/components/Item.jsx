import React from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
} from "react-native";
import PropTypes from "prop-types";
import { theme } from "../constants";
import Typography from "./Typography";
import Dot from "./Dot";

const { width } = Dimensions.get("screen");

const Item = ({
  text,
  removeItem,
  onSelect,
}) => {
  return (
    <TouchableOpacity
      testID={"button1"}
      onPress={onSelect}
      style={styles.container}
    >
      <Dot />
      <Typography
        color={theme.colors.seconderyTextColor}
        align="left"
        weight="avarage"
        style={styles.text}
      >
        {text}
      </Typography>
      <TouchableOpacity
        testID={"button2"}
        onPress={removeItem}
        style={styles.button}
      >
        <Typography
          color={theme.colors.seconderyTextColor}
          size="h6"
          weight="avarage"
        >
          REMOVE
        </Typography>
      </TouchableOpacity>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    backgroundColor: theme.colors.white,
    margin: 8,
    borderRadius: 24,
    padding: 16,
  },
  button: {
    width: 80,
    padding: 8,
    zIndex: 99999,
  },
  text: {
    marginLeft: 8,
    width: width * 0.6,
  },
  textContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
});

Item.propTypes = {
  text: PropTypes.string.isRequired,
  removeItem: PropTypes.func,
  onSelect: PropTypes.func,
};

export default Item;
