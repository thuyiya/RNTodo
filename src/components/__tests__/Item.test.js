import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react-native";
import Item from "../Item";

describe("<Item />", () => {
  const item = {
    id: "1",
    text: "Hellow",
  };

  it("has 3 children", () => {
    const tree = renderer.create(<Item {...item} />).toJSON();
    expect(tree.children.length).toBe(3);
  });

  it(`displays the item text as ${item.text}`, () => {
    render(<Item {...item} />);
    expect(screen.queryByText(item.text)).toBeTruthy();
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Item {...item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
