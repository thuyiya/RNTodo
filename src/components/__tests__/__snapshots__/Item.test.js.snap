// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 24,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 8,
      "opacity": 1,
      "padding": 16,
    }
  }
  testID="button1"
>
  <View
    style={
      {
        "backgroundColor": "#0145a6",
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      }
    }
  />
  <Text
    style={
      [
        {
          "color": "#787878",
          "fontSize": 16,
          "fontWeight": "400",
          "margin": 0,
          "textAlign": "left",
        },
        {
          "marginLeft": 8,
          "width": 450,
        },
      ]
    }
  >
    Hellow
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "padding": 8,
        "width": 80,
        "zIndex": 99999,
      }
    }
    testID="button2"
  >
    <Text
      style={
        [
          {
            "color": "#787878",
            "fontSize": 14,
            "fontWeight": "400",
            "margin": 0,
            "textAlign": "center",
          },
          {},
        ]
      }
    >
      REMOVE
    </Text>
  </View>
</View>
`;
